src/Common/AsyncLoader.cpp

```diff
@@ -275,7 +268,33 @@ void AsyncLoader::wait()
     }
 }
 
-void AsyncLoader::stop()
+void AsyncLoader::shutdown()
+{
+    {
+        std::unique_lock lock{mutex};
+        shutdown_requested = true;
+        is_running = false;
+
+        while (!scheduled_jobs.empty())
+        {
+            const auto it = scheduled_jobs.begin();
+            if (!it->second.isExecuting())
+            {
+                LoadJobPtr job = it->first;
+                auto e = std::make_exception_ptr(Exception(ErrorCodes::ASYNC_LOAD_CANCELED, "AsyncLoader was shut down"));
+                finish(job, LoadStatus::CANCELED, e, lock);
+                chassert(lock.owns_lock());
+                chassert(!scheduled_jobs.contains(job));
+            }
+        }
+    }
```

Comment:
Maybe we could reuse AsyncLoader::remove() to remove all jobs. It now starts looking very similar